component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java

```diff
@@ -42,12 +51,43 @@
 public class PasswordChangeHandler extends AbstractEventHandler implements IdentityConnectorConfig {
     private static final Log log = LogFactory.getLog(PasswordChangeHandler.class);
 
+
     @Override
     public void handleEvent(Event event) throws IdentityEventException {
+
+        String eventName = event.getEventName();
         // Fetching event properties
         String username = (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.USER_NAME);
         UserStoreManager userStoreManager = (UserStoreManager) event.getEventProperties()
                 .get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+
+        //password grant handler - password expiry validation
+        if (eventName.equals(PasswordPolicyConstants.PASSWORD_GRANT_POST_AUTHENTICATION_EVENT)) {
```

Comment:
better put the constant first. PasswordPolicyConstants.PASSWORD_GRANT_POST_AUTHENTICATION_EVENT.equals(eventName)